src/main/java/algorithms/SumAllIntegersInString.java

```diff
@@ -0,0 +1,41 @@
+package algorithms;
+
+public class SumAllIntegersInString {
+    public static int sumAllIntsInString0(String str) {
+        int sum = 0;
+        int currNum = 0;
+        int sign = 1;
+        for (int i = 0; i < str.length(); i++) {
+            char ch = str.charAt(i);
+            if (ch == '-') {
+                if (currNum != 0) {
```

Comment:
We may not need this if-statement in this case.

Reply:
By line 11 or line 10? If it's line 11, we can indeed get rid of it.

Reply:
Yeah, it is 11; as, even if 'currNum' is '0' it will not affect.

Reply:
Updated.